client/data/payment-activity/types.d.ts

```diff
@@ -24,12 +24,12 @@ export interface PaymentActivityData {
 }
 
 export interface PaymentActivityState {
-	paymentActivityData?: PaymentActivityData;
-	isLoading?: boolean;
+	[ key: string ]: PaymentActivityData;
```

Comment:
Do we cache each preset's data in the reducer under the preset name? Is that necessary, or an optimisation?

Would this cache ever go stale?

Might be worth a comment to document the intended state shape and how it's supposed to work. This react state management code can be hard to read :) 

Do we want a separate `isLoading` flag for each preset's data? What does that give us?

Reply:
> Do we cache each preset's data in the reducer under the preset name? Is that necessary, or an optimisation?

It is cached under a serialisation of the query object using `client/utils/data` `getResourceId()`, which is a pattern used for other query response caching, e.g. transactions, disputes. So rather than just the preset name, it will include all query params: currency, timezone, date_start, date_end.

https://github.com/Automattic/woocommerce-payments/blob/b18699c5940009ac8eb37defb202792af331d2bd/client/utils/data.js#L3-L5

<img width="396" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/3147296/0a04de9b-4833-40ad-8f8a-2d43754643ce">

Reply:
> Would this cache ever go stale?

Yes, this would be stale if you left the Payments → Overview screen open for 24 hours and didn't refresh the page. 

I assume this would also be the case for all of our wp.data stores (list screens, account status, etc) – unsure if this is an issue or not since the cache is not preserved on page refresh.

Reply:
Sounds good, I had a vague assumption there might be a "midnight" bug, which seems very minor / low risk 🚀 

Thanks for explaining, it might be worth adding comments about this to the data layer to explain the intentions of the caching, make it more obvious for future engineers.

Reply:
In 1cd45f7 I've added a comment at the point of storing the payment activity data (the payment-activity reducer) – I think this is helpful, what do you think @haszari? 

https://github.com/Automattic/woocommerce-payments/blob/1cd45f7e11c26136a7b6795f768371481e868967/client/data/payment-activity/reducer.ts#L18-L28

Reply:
❓ For my understanding - Consider this scenarion - 
If the merchant switches to `Today` or any of the `x to date` option , checks a few different other presets and comes back. During this time presuming it is a high volume website and it gets a few transactions, would the value we see in `Today` remain same as earlier until we refresh the browser?

Reply:
Will it be better to remove `.set( { hour: 23, minute: 59, second: 59, millisecond: 0 } );` from [todayEndOfDay](https://github.com/Automattic/woocommerce-payments/blob/cdfb1ebae0b6594f4eee841bec652daf93de0c7a/client/components/payment-activity/hooks.ts#L35-L36) ?

Reply:
> During this time presuming it is a high volume website and it gets a few transactions, would the value we see in Today remain same as earlier until we refresh the browser?

Yep you are right, if the query hasn't changed (e.g. `date_start`, `date_end`), then the cached value will be shown until the browser is refreshed.

This is good for the general case when a user is comparing date ranges → doesn't have to wait for network requests that were made only a few seconds ago.

We could also have the best of both worlds by showing the stale data while we invalidate and re-fetch in the background → update the values on response:
- after X seconds/hours/days
- on browser/tab re-focus
- if a webhook is received (transaction, dispute, etc)

I see reporting as a compelling opportunity to pursue a more aggressive "live" caching strategy since it may add value – v2 will be a great time to align on a good approach and implement this.

But I don't think we need this in this PR, considering that existing WooPayments screens and any PHP-driven WP-admin screens also show stale data until the page is refreshed.

Reply:
Thanks Eric!  I am leaving this as-it-is for now. I have created an issue to discuss this further and change things if required - https://github.com/Automattic/woocommerce-payments/issues/8944

Reply:
Looks good, thanks Eric (& great discussion for future Nagesh!).

It might be worth clarifying in the comment what we are using for the key, i.e. our chosen caching strategy.